### PR TITLE
Re-Former Project - Microverse Re-Former Rails Project #4

### PR DESCRIPTION
**Pull request by JohnHernCode**
Re-Former project in Rails #4 - Forms usage in Rails
Milestones:
1. Create HTML Form
2. Create Form_Tag Form
3. Create Form_With Form
4. Add Edit Functionality

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (kapkev) [inline app/controllers/users_controller.rb]: - [x] Please remove commented code!

**Review by kapkev — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻
### Well done on making the previously required changes, your project is almost done, just a few things before we move on:

- [x] You have commented code, please remove it;
- [x] Currently after save a new user, your project isn't redirecting me to the new user path, where I should be able to edit it, please fix that;

After making the required changes, please submit it again.
<hr>
If you have any question, feel free to reach me through slack

**Comment by jpdf00** [comment]
Just to leave it registered:
 - TSE asked to remove commented code but the project specification asked to comment code and to leave it there.
 - TSE asked for the "create new user" button to redirect to the edit page of this recently create user. This isn't part of the project description and shouldn't be asked.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      redirect_to edit_user_path(@user)
```
Comment (enaburekhan) [inline app/controllers/users_controller.rb]: change line 10 to redirect_to new_user_path(@user)

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+
+    if @user.update(user_params)
+      redirect_to new_user_path
```
Comment (enaburekhan) [inline app/controllers/users_controller.rb]: change line 24 to:  redirect_to edit_user_path

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Required changes:full_moon:
**Hi team!
Great job so far on this project:clap::clap:. The codebase looks very clean. However, we still need to make a few changes to get this project approved:muscle:.**

**Mandatory requirements**

- [x] Create action

- redirect_to new_user_path if user was saved ( [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_create.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)). As already highlighted in the PR below. Fix this

- [x] update action

- redirect_to edit_user_path if user was saved. Also correct this.

- [x] Add the commented out code in new.html.erb in view. The previous Tse only required you to remove the commented out code in the controller and not view. It was clearly highlighted in the PR.

[ Optional]
- Proofread the readme especially the Usage section to make corrections.  Where you wrote rails console. I'm sure you meant rails server

Make the above-required changes and when you are ready, request for another review.
Kindly drop your questions :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy Coding:blush:
Eric Enaburekhan

**Review by rindrajosia — APPROVED** [review]
Status :100:  Approved :1st_place_medal:
Great work!! Congrats, the project is approved :heavy_check_mark:

[ Optional]

- [ ]  In your readme, `Usage section`, before `rails db:migrate` add `rails db:create`
- [ ] Partial created for user form and reused in edit and new view
- [ ] Form view to displays a list of the error messages (with #errors and full_messages usage) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_extra_shared_error.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
- [ ] Validate the email format by using either a regular expression or a built in URI method.

Keep up the good work :muscle:

Rindra Josia

**Pull request merged**